packages/storage-ui/src/Components/Modules/NFTsList/NFTItem.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from "react"
+import { makeStyles, createStyles } from "@chainsafe/common-theme"
+import {
+  Typography,
+  Button
+} from "@chainsafe/common-components"
+import { CSSTheme } from "../../../Themes/types"
+import { Trans } from "@lingui/macro"
+import clsx from "clsx"
+import axios from "axios"
+import { trimChar } from "../../../Utils/pathUtils"
+
+const IPFS_GATEWAY = process.env.REACT_APP_IPFS_GATEWAY || "https://ipfs.io/ipfs/"
+
+const useStyles = makeStyles(({ constants, palette }: CSSTheme) =>
+  createStyles({
+    root: {
+      maxWidth: "100%"
+    },
+    imageBox: {
+      width: "100%",
+      objectFit: "cover",
+      marginBottom: constants.generalUnit
+    },
+    nameTitle: {
+      marginBottom: constants.generalUnit * 0.5
+    },
+    nameContainer: {
+      borderBottom: `1px solid ${palette.additional["gray"][6]}`,
+      marginBottom: constants.generalUnit * 0.5
+    },
+    cidRow: {
+      display: "flex",
+      alignItems: "center",
+      justifyContent: "space-between",
+      "&.clickable": {
+        cursor: "pointer"
+      }
+    },
+    cidStartSection: {
+      display: "table",
+      tableLayout: "fixed",
+      width: "100%",
+      whiteSpace: "nowrap"
+    },
+    cidSubtitle: {
+      width: "35px",
+      display: "table-cell",
+      color: palette.additional["gray"][7]
+    },
+    cid: {
+      display: "table-cell",
+      overflow: "hidden",
+      textOverflow: "ellipsis"
+    },
+    copyButton: {
+      marginLeft: constants.generalUnit,
+      color: palette.primary.main,
+      padding: `${constants.generalUnit * 0.5}px 0 !important`
+    },
+    copiedText: {
+      marginLeft: constants.generalUnit,
+      padding: `${constants.generalUnit * 0.5}px 0 !important`
+    }
+  })
+)
+
+interface NFTData {
+  name: string
+  CID: string
+  image?: string
+}
+
+const NFTItem = ({ CID }: {CID: string}) => {
+  const classes = useStyles()
+  const [isCopied, setIsCopied] = useState(false)
+  const [NFTData, setNFTData] = useState<NFTData>()
```

Comment:
I'm very surprised that TS doesn't complain here, I guess it should be 
```suggestion
  const [NFTData, setNFTData] = useState<NFTData | undefined>()
```